### PR TITLE
CAMS-356 functions v4 upgrade

### PR DESCRIPTION
**Pull request by amorrow-flexion**
# Purpose

upgrade our Node API Azure Functions version from 3.x to 4.x

# Major Changes

- install @azure/functions@4 npm package
- update function definitions and tests to use new version
- update http modules to properly handle new request and response types

# Testing/Validation

- Ran application locally and validated application functionality
- deployed application to a dev environment ****

# Notes

- Brian's experimental PR

# Definition of Done:

- [ ] Acceptance criteria met - The system meets all acceptance criteria
- [ ] Feature toggles created - Features that are deployed but not released have toggles
- [ ] Design QA passed - All visual UI looks and works the way it’s supposed to
- [ ] Usability validated - Team ensures the working code is easy to use
- [ ] Accessibility tests passed - System passes automated and manual accessibility tests and supports applicable devices
- [ ] Code refactored for clarity - Developers can understand the work simply by reviewing the code
- [ ] Dependency Rule followed - More important code doesn’t directly depend on less important code
- [ ] Development debt eliminated - UX and code aligns to the team’s latest understanding of the domain
- [ ] Source code merged - Code has been merged into the main branch
- [ ] 90% unit test coverage achieved - Automated test coverage tools indicate coverage of >= 90%
- [ ] Code reviewed - Code is reviewed by at least two other team members before being merged
- [ ] Code quality checks passed - Code passes all automated quality checks
- [ ] Security scans passed - Code passes expected scans for vulnerabilities and compliance issues
- [ ] Threat model updated - The threat model incorporates new threats and remediations
- [ ] Build process updated - Automated build process include code supporting the user story
- [ ] Load/performance tests passed - Performance tests include the functionality of the user story
- [ ] Documents are updated - All required documentation is up to date and version controlled
- [ ] Code is deployed - Code is deployed to the highest production like environment

### CODE REVIEW

File: backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+describe('Attorneys Controller Tests', () => {
+  test('', () => {});
```
Comment (btposey) [inline backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts:2]: No tests??

File: backend/functions/consolidations/consolidations.function.ts [hunk 2]
```diff
@@ -25,24 +29,18 @@ export default async function handler(
     );
     const controller = new OrdersController(applicationContext);
     const procedure = applicationContext.request.params.procedure;
-    let response;
+    let response: ManageConsolidationResponse;
 
     if (procedure === 'reject') {
```
Comment (btposey) [inline backend/functions/consolidations/consolidations.function.ts:34]: When delegating handling the request to the controller from the function I would have had a single, consistently named function on each controller that would accept the CamsApplicationContext that would then have the routing logic in it, rather than having to inspect the request and contain logic in the Azure function.

**Review by btposey — CHANGES_REQUESTED** [review]

File: backend/functions/lib/adapters/types/basic.d.ts [hunk 3]
```diff
@@ -11,12 +11,21 @@ export interface LoggerHelper {
   camsError: (error: CamsError) => void;
 }
 
-// TODO: By extending Context from Azure Functions we have allowed Azure to invade this codebase.
-export interface ApplicationContext extends Context {
+// // TODO: By extending Context from Azure Functions we have allowed Azure to invade this codebase.
```
Comment (btposey) [inline backend/functions/lib/adapters/types/basic.d.ts:14]: Remove this commented code

**Review by btposey — CHANGES_REQUESTED** [review]

File: backend/functions/case-assignments/case.assignment.function.test.ts [hunk 4]
```diff
@@ -1,243 +1,178 @@
...
-    };
-    const expectedResponse = { message: 'caseId must be formatted like 01-12345.', success: false };
-
-    const httpErrorSpy = jest.spyOn(httpResponseModule, 'httpError');
-    await httpTrigger(context, requestOverride);
-    expect(context.res.body).toEqual(expectedResponse);
-    expect(context.res.statusCode).toEqual(400);
-    expect(httpErrorSpy).toHaveBeenCalledWith(expect.any(AssignmentError));
-    expect(httpErrorSpy).not.toHaveBeenCalledWith(expect.any(UnknownError));
+  afterEach(() => {
```
Comment (jamesobrooks) [inline backend/functions/case-assignments/case.assignment.function.test.ts:48]: Add newline:
```suggestion

  afterEach(() => {
```

File: backend/functions/case-assignments/case.assignment.function.test.ts [hunk 5]
```diff
@@ -1,243 +1,178 @@
...
+    jest.clearAllMocks();
   });
+  test('Return the function response with the assignment Id created for the new case assignment', async () => {
```
Comment (jamesobrooks) [inline backend/functions/case-assignments/case.assignment.function.test.ts:51]: Add newline:
```suggestion

  test('Return the function response with the assignment Id created for the new case assignment', async () => {
```

File: backend/functions/case-history/case-history.function.test.ts [hunk 6]
```diff
@@ -1,59 +1,87 @@
...
+describe('Case History Function Tests', () => {
+  const defaultRequestProps: Partial<CamsHttpRequest> = {
+    method: 'GET',
     params: {
       caseId: '',
     },
-    method: 'GET',
+  };
+  let context;
+  beforeEach(() => {
```
Comment (jamesobrooks) [inline backend/functions/case-history/case-history.function.test.ts:27]: Add newline:
```suggestion

  beforeEach(() => {
```

File: backend/functions/cases/cases.function.ts [hunk 7]
```diff
@@ -1,57 +1,46 @@
...
-      responseBody = await casesController.getCaseDetails({
-        caseId: request.params.caseId,
+    const applicationContext = await ContextCreator.applicationContextCreator(
+      invocationContext,
+      request,
+      logger,
+    );
+    const casesController = new CasesController(applicationContext);
+
+    if (request.method === 'GET' && applicationContext.request.params.caseId) {
```
Comment (jamesobrooks) [inline backend/functions/cases/cases.function.ts:27]:
```suggestion
    if (applicationContext.request.method === 'GET' && applicationContext.request.params.caseId) {
```

File: backend/functions/consolidations/consolidations.function.ts (hunk 2 above)
Comment (jamesobrooks) [reply backend/functions/consolidations/consolidations.function.ts:34]: No objections from me. But not in this PR preferably.

File: backend/functions/lib/adapters/gateways/cases.cosmosdb.repository.ts [hunk 8]
```diff
@@ -104,6 +105,7 @@ export class CasesCosmosDbRepository implements CasesRepository {
       ],
     };
     const response = await this.queryData<CaseHistory>(context, querySpec);
+    console.log('Case history repo:', caseId, query, response);
```
Comment (jamesobrooks) [inline backend/functions/lib/adapters/gateways/cases.cosmosdb.repository.ts:108]: Remove.
```suggestion
```

File: backend/functions/lib/adapters/utils/http-response.test.ts [hunk 9]
```diff
@@ -1,34 +1,31 @@
...
+type TestType = {
+  testString: string;
+};
 
 describe('Tests out http responses', () => {
-  test('Should return properly formatted http success response', () => {
-    const expectedBody = { testObject: 'testValue' };
-    const actualResult = httpSuccess(expectedBody);
+  test('Should return properly formatted http success response', async () => {
+    // const expectedBody = { body: { data: { testObject: 'testValue' } };
```
Comment (jamesobrooks) [inline backend/functions/lib/adapters/utils/http-response.test.ts:9]: Remove.
```suggestion
```

File: backend/functions/azure/application-context-creator.ts [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+
+function getLogger(invocationContext: InvocationContext) {
+  /* eslint-disable-next-line @typescript-eslint/no-explicit-any */
+  const logWrapper: Console['log'] = (...args: any[]) => {
+    invocationContext.log(args);
+  };
+  return new LoggerImpl(invocationContext.invocationId, logWrapper);
+}
+
+// TODO: consider switching to an object since the logger and request arguments are
```
Comment (btposey) [inline backend/functions/azure/application-context-creator.ts:22]: Converting the args to an options object is pretty easy. We should do this.

File: backend/functions/case-assignments/case.assignment.function.ts [hunk 11]
```diff
@@ -1,45 +1,47 @@
...
   try {
     applicationContext.session =
       await ContextCreator.getApplicationContextSession(applicationContext);
-
+    let assignmentResponse: HttpResponseInit;
     if (request.method === 'POST') {
-      const listOfAttorneyNames = request.body.attorneyList;
-      const role = request.body.role;
-      await handlePostMethod(applicationContext, request.body.caseId, listOfAttorneyNames, role);
+      //We should be doing this in the controller
```
Comment (btposey) [inline backend/functions/case-assignments/case.assignment.function.ts:27]: `// We should be doing this in the controller`

File: backend/functions/case-associated/case-associated.function.test.ts [hunk 12]
```diff
@@ -1,43 +1,63 @@
...
+import { InvocationContext } from '@azure/functions';
+import {
+  buildTestResponseError,
+  buildTestResponseSuccess,
+  createMockAzureFunctionRequest,
+} from '../azure/testing-helpers';
+import { EventCaseReference } from '../../../common/src/cams/events';
 
 describe('Case summary function', () => {
+  jest.spyOn(ContextCreator, 'getApplicationContextSession').mockResolvedValue(
```
Comment (btposey) [inline backend/functions/case-associated/case-associated.function.test.ts:17]: This would be a good testing utility function.

File: backend/functions/lib/controllers/case-assignment/case.assignment.controller.test.ts [hunk 13]
```diff
@@ -11,6 +14,9 @@ import {
 import { CamsRole } from '../../../../../common/src/cams/roles';
 import { CamsUserReference } from '../../../../../common/src/cams/users';
 import { MANHATTAN } from '../../../../../common/src/cams/test-utilities/offices.mock';
+import { UnknownError } from '../../common-errors/unknown-error';
+import HttpStatusCodes from '../../../../../common/src/api/http-status-codes';
+import { buildTestResponseSuccess } from '../../../azure/testing-helpers';
```
Comment (jamesobrooks) [inline backend/functions/lib/controllers/case-assignment/case.assignment.controller.test.ts:19]: We should not be reaching into `/backend/functions/azure` from anywhere in `/lib`.

File: backend/functions/case-assignments/case.assignment.function.test.ts [hunk 14]
```diff
@@ -1,243 +1,178 @@
...
+        ...defaultRequestProps,
+        ...requestOverride,
+      });
+      const { azureHttpResponse } = buildTestResponseError(error);
+      jest
+        .spyOn(CaseAssignmentController.prototype, 'createTrialAttorneyAssignments')
+        .mockRejectedValue(error);
+
+      const response = await handler(request, context);
+      console.log('Response:   ', response);
```
Comment (fmaddenflx) [inline backend/functions/case-assignments/case.assignment.function.test.ts:101]: remove console log

File: backend/functions/case-assignments/case.assignment.function.test.ts [hunk 15]
```diff
@@ -1,243 +1,178 @@
...
+    [
+      'invalid role',
+      '001-90-90123',
+      'TrialDragon',
+      'Invalid role for the attorney. Requires role to be a TrialAttorney for case assignment.',
+    ],
+  ];
+  test.each(errorTestCases)(
+    'should return proper error response for %s',
+    async (_caseName: string, caseId: string, role: string, message: string) => {
```
Comment (fmaddenflx) [inline backend/functions/case-assignments/case.assignment.function.test.ts:82]: Why are we even defining a caseName if we're not going to use it.  Is it just for reference sake so we understand what the test is for?   Oh... is that was is being used in the string template with %s?

File: backend/functions/case-associated/case-associated.function.ts [hunk 16]
```diff
@@ -1,37 +1,40 @@
...
-import ContextCreator from '../lib/adapters/utils/application-context-creator';
+import { app, HttpRequest, HttpResponseInit, InvocationContext } from '@azure/functions';
+import ContextCreator from '../azure/application-context-creator';
 import { initializeApplicationInsights } from '../azure/app-insights';
 import { CaseAssociatedController } from '../lib/controllers/case-associated/case-associated.controller';
+import { toAzureError, toAzureSuccess } from '../azure/functions';
 
 dotenv.config();
 
+const MODULE_NAME = 'CASE-ASSOCIATED-FUNCTION' as const;
```
Comment (fmaddenflx) [inline backend/functions/case-associated/case-associated.function.ts:10]: What is the purpose of typecasting this as a const if it's being defined as a const?

File: backend/functions/lib/use-cases/case-management.test.ts [hunk 17]
```diff
@@ -219,13 +221,7 @@ describe('Case management tests', () => {
         },
       ];
 
-      // TODO: This is gross. The response wrapper should be added by the controller, not the use case.
-      // I Agree.  This is gross.
```
Comment (btposey) [inline backend/functions/lib/use-cases/case-management.test.ts:223]: Still gross. Still should be migrated to the controller. Can punt to another day since I'm certain this isn't the last time we will be working in this use case.

File: backend/functions/orders-sync/orders-sync.function.ts [hunk 18]
```diff
@@ -1,23 +1,33 @@
...
-import ContextCreator from '../lib/adapters/utils/application-context-creator';
+import { app, InvocationContext, Timer } from '@azure/functions';
+import ContextCreator from '../azure/application-context-creator';
 import { initializeApplicationInsights } from '../azure/app-insights';
 import { OrdersController } from '../lib/controllers/orders/orders.controller';
 
 import * as dotenv from 'dotenv';
+import { toAzureError } from '../azure/functions';
 
 // TODO: We need to look into upgrading this to use v4 of Azure Functions
```
Comment (btposey) [inline backend/functions/orders-sync/orders-sync.function.ts:10]: `// TODO: We need to look into upgrading this to use v4 of Azure Functions
Comment` Is this still something we need to look into? Is this talking about the use of `dotenv` or the function in general?

File: backend/functions/orders-sync/orders-sync.function.ts [hunk 19]
```diff
@@ -1,23 +1,33 @@
 // TODO: The original context was InvocationContext
```
Comment (btposey) [inline backend/functions/orders-sync/orders-sync.function.ts:1]: Delete

File: backend/functions/lib/controllers/orders/orders.controller.ts [hunk 20]
```diff
@@ -58,14 +60,17 @@ export class OrdersController {
 
   public async getSuggestedCases(
     context: ApplicationContext,
-    caseId: string,
-  ): Promise<GetSuggestedCasesResponse> {
+  ): Promise<CamsHttpResponseInit<CaseSummary[]>> {
```
Comment (jamesobrooks) [inline backend/functions/lib/controllers/orders/orders.controller.ts:63]: Following the other functions, this one would be:
```suggestion
  ): Promise<GetSuggestedCasesResponse> {
```

File: backend/functions/cases/cases.function.test.ts [hunk 21]
```diff
@@ -1,65 +1,122 @@
...
+      name: 'searchCases',
+      error: new Error('test-error'),
+      requestParam: {
+        method: 'POST',
+        body: { caseNumber: '00-12345' },
+      },
+    },
+  ];
+
+  test.each(params)(`should handle error from $name`, async (params: TestParams) => {
```
Comment (fmaddenflx) [inline backend/functions/cases/cases.function.test.ts:112]: This is interesting.  In another test we were using the %s in a template string for test.each.  Here we're using $name.  I guess using $foo is possible because we're using an array of objects here?

Comment (fmaddenflx) [reply backend/functions/cases/cases.function.test.ts:112]: Have we verified that these tests are printing the expected titles in both cases?

File: backend/functions/cases/cases.function.ts [hunk 22]
```diff
@@ -1,57 +1,46 @@
...
 import { initializeApplicationInsights } from '../azure/app-insights';
-import { CaseBasics } from '../../../common/src/cams/cases';
-import { ResponseBody } from '../../../common/src/api/response';
-import { httpRequestToCamsHttpRequest } from '../azure/functions';
+import { toAzureError, toAzureSuccess } from '../azure/functions';
 
 dotenv.config();
 
-initializeApplicationInsights();
+const MODULE_NAME = 'CASES-FUNCTION' as const;
```
Comment (fmaddenflx) [inline backend/functions/cases/cases.function.ts:10]: Once again.  I'm curious as to what this buys us using "as const" here?

File: user-interface/src/lib/hooks/UseApi2.ts [hunk 23]
```diff
@@ -1,67 +1,9 @@
...
-  searchCases,
-};
-
-// TODO: Deprecate this hook.
 export function useApi2() {
-  return Api2;
+  return Api2Module.Api2;
 }
 
+// These exports are here for legacy support. These can be removed once these are no longer imported from the hook source file.
```
Comment (btposey) [inline user-interface/src/lib/hooks/UseApi2.ts:7]: Need to check if `Api2` is still imported from the hook source file anywhere else.

File: backend/functions/case-associated/case-associated.function.ts (hunk 16 above)
Comment (btposey) [reply backend/functions/case-associated/case-associated.function.ts:10]: It is a const^2

File: backend/functions/me/me.function.ts [hunk 24]
```diff
@@ -1,20 +1,37 @@
...
+    request,
+  );
   try {
-    context.session = await ContextCreator.getApplicationContextSession(context);
-    const response = { success: true, body: context.session };
-    functionContext.res = httpSuccess(response);
-  } catch (camsError) {
-    context.logger.camsError(camsError);
-    functionContext.res = httpError(camsError);
+    applicationContext.session =
```
[the remark this answers is not in the document]
Comment (jamesobrooks) [reply backend/functions/me/me.function.ts:17]: Just a note that the context in this case does not actually need the logger added. We could still add it though, because this may be the only function that would never need it, and the logger argument being optional could present bugs in the future.

File: backend/functions/offices/offices.function.test.ts [hunk 25]
```diff
@@ -1,27 +1,31 @@
-import httpTrigger from '../offices/offices.function';
 import { CamsError } from '../lib/common-errors/cams-error';
-import { createMockAzureFunctionRequest } from '../azure/functions';
-import ContextCreator from '../lib/adapters/utils/application-context-creator';
+import ContextCreator from '../azure/application-context-creator';
```
Comment (jamesobrooks) [inline backend/functions/offices/offices.function.test.ts:2]: We should not be reaching into `/backend/functions/azure` from anywhere in `/lib`.

File: backend/functions/orders-manual-sync/orders-manual-sync.function.ts [hunk 26]
```diff
@@ -19,23 +21,30 @@ initializeApplicationInsights();
...
+  });
 
   const ordersController = new OrdersController(context);
   try {
-    const results = await ordersController.syncOrders(context, request.body);
-    functionContext.res = httpSuccess(results);
-  } catch (camsError) {
-    context.logger.camsError(camsError);
-    functionContext.res = httpError(camsError);
+    const results = await ordersController.syncOrders(context, request.body as SyncOrdersOptions);
```
Comment (jamesobrooks) [inline backend/functions/orders-manual-sync/orders-manual-sync.function.ts:40]: We should only be sending the context to the controller, not dealing with the request.

File: backend/functions/orders-manual-sync/orders-manual-sync.function.ts [hunk 27]
```diff
@@ -19,23 +21,30 @@ initializeApplicationInsights();
...
+    return toAzureSuccess(results);
+  } catch (error) {
+    return toAzureError(context, MODULE_NAME, error);
   }
-};
-
-export default httpTrigger;
+}
+app.http('order-manual-sync', {
```
Comment (jamesobrooks) [inline backend/functions/orders-manual-sync/orders-manual-sync.function.ts:46]: Add newline:
```suggestion

app.http('order-manual-sync', {
```

File: backend/functions/orders-suggestions/orders-suggestions.function.test.ts [hunk 28]
```diff
@@ -1,53 +1,50 @@
...
       method: 'GET',
-    };
-    const expectedErrorResponse = {
-      success: false,
-      message: 'Mocked error',
-    };
-    await httpTrigger(context, request);
-    expect(context.res.body).toMatchObject(expectedErrorResponse);
+    });
+    console.log('Azure Http Response:    ', azureHttpResponse);
```
Comment (jamesobrooks) [inline backend/functions/orders-suggestions/orders-suggestions.function.test.ts:45]: Remove.
```suggestion
```

File: backend/functions/orders-sync/orders-sync.function.test.ts [hunk 29]
```diff
@@ -25,13 +27,25 @@ const syncResponse: SyncOrdersStatus = {
 
 describe('Orders Sync Function tests', () => {
   /* eslint-disable-next-line @typescript-eslint/no-require-imports */
```
Comment (jamesobrooks) [inline backend/functions/orders-sync/orders-sync.function.test.ts:29]: Remove.
```suggestion
```

File: backend/functions/orders-sync/orders-sync.function.ts (hunk 18 above)
Comment (jamesobrooks) [reply backend/functions/orders-sync/orders-sync.function.ts:10]: I don't think they have anything to do with one another. I'd be happy to get rid of that dependency if we don't need it though.

File: backend/functions/orders-sync/orders-sync.function.ts [hunk 30]
```diff
@@ -1,23 +1,33 @@
...
-  } catch (camsError) {
-    context.logger.camsError(camsError);
+    const appContext = await ContextCreator.getApplicationContext({ invocationContext, logger });
+    const ordersController = new OrdersController(appContext);
+    await ordersController.syncOrders(appContext);
+  } catch (error) {
+    toAzureError(logger, MODULE_NAME, error);
   }
 }
+app.timer('orders-sync', {
```
Comment (jamesobrooks) [inline backend/functions/orders-sync/orders-sync.function.ts:30]: Add newline:
```suggestion

app.timer('orders-sync', {
```

File: backend/functions/orders/orders.function.ts [hunk 31]
```diff
@@ -1,65 +1,49 @@
...
+      const orderGet = await orderController.getOrders(context);
+      response = toAzureSuccess<Order[]>(orderGet);
+    } else if (context.request.method === 'PATCH') {
+      //TODO: Json Mapping with these requestBody objects
+      const id = context.request.params['id'];
+      const camsResponse = await orderController.updateOrder(
+        context,
+        id,
+        context.request.body as TransferOrderAction,
+      );
```
Comment (jamesobrooks) [inline backend/functions/orders/orders.function.ts:35]: Two comments regarding this TODO. First, we should not be splitting the request parameters out anymore before calling the controller. Second, if the mapping idea is to be able to turn `context.request.params['id']` into `context.request.params.id`, we should already be able to do that (for example see `orders.controller.ts`).

File: backend/functions/lib/adapters/gateways/cases.cosmosdb.repository.ts (hunk 8 above)
Comment (fmaddenflx) [inline backend/functions/lib/adapters/gateways/cases.cosmosdb.repository.ts:108]: do we need this console log?

File: docs/architecture/diagrams/cams-containers.md
Comment (jamesobrooks) [inline docs/architecture/diagrams/cams-containers.md:1]: This is not right!

File: backend/functions/lib/adapters/utils/http-response.test.ts (hunk 9 above)
Comment (fmaddenflx) [inline backend/functions/lib/adapters/utils/http-response.test.ts:9]: looks like we can remove this commented code for expectedBody.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx [hunk 32]
```diff
@@ -102,11 +58,18 @@ describe('ConsolidationOrderAccordion tests', () => {
       'consolidations-enabled': true,
     };
     vitest.spyOn(FeatureFlagHook, 'default').mockReturnValue(mockFeatureFlags);
-    setupApiGetMock();
+    vitest
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:61]: I've noticed that some files have `vitest.spyOn` rather than `vi.spyOn` which generally leads to new spies being written with vitest. We should probably change them all. This could be in a different PR.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx [hunk 33]
```diff
@@ -267,10 +230,11 @@ describe('ConsolidationOrderAccordion tests', () => {
     });
   });
 
-  test('should correctly enable/disable buttons when selecting consolidated cases and lead case from order case list table', async () => {
+  test.skip('should correctly enable/disable buttons when selecting consolidated cases and lead case from order case list table', async () => {
```
Comment (jamesobrooks) [inline user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:233]: We should unskip or delete this test if no longer needed.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx (hunk 32 above)
Comment (jamesobrooks) [inline user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:61]: We should probably replace `vitest.spyOn` with `vi.spyOn` across all tests. But probably in a subsequent PR.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx
Comment (jamesobrooks) [inline user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:1]: The skipped tests here are what used to provide coverage for the whole directory. They may prove useful for writing the tests of the controls and store to see what we were doing before, but they may not so could probably be removed.

File: backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts (hunk 1 above)
Comment (fmaddenflx) [reply backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts:2]: lol woops.

File: backend/functions/lib/controllers/case-history/case-history.controller.test.ts [hunk 34]
```diff
@@ -8,42 +8,55 @@ import {
...
     const caseId = NORMAL_CASE_ID;
+    const request: CamsHttpRequest = mockCamsHttpRequest({
+      params: { caseId },
+    });
+    applicationContext.request = request;
     const controller = new CaseHistoryController(applicationContext);
-    const result = await controller.getCaseHistory(applicationContext, { caseId });
-    expect(result.isSuccess).toBeTruthy();
-    expect(result['data']).toEqual(CASE_HISTORY);
+    const result = await controller.getCaseHistory(applicationContext, request);
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/case-history/case-history.controller.test.ts:29]: We can probably update this, since request is probably being passed in within the applicationContext, the controller or use case should grab it off the applicationContext.

File: backend/functions/lib/controllers/case-history/case-history.controller.ts [hunk 35]
```diff
@@ -20,22 +16,21 @@ export class CaseHistoryController {
...
-  ): Promise<ResponseBody<CaseHistory[]>> {
+    request: CamsHttpRequest,
+  ): Promise<CamsHttpResponseInit<CaseHistory[]>> {
     try {
-      const caseHistory = await this.useCase.getCaseHistory(context, request.caseId);
-      return {
-        meta: {
-          isPaginated: false,
-          self: context.req.url,
+      const caseHistory = await this.useCase.getCaseHistory(context, request.params.id);
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/case-history/case-history.controller.ts:22]: same here.  The request is embedded in the context, so we don't need to pass the id to getCaseHistory.

File: backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts [hunk 36]
```diff
@@ -1,40 +1,46 @@
...
   });
 
   test('should return success if case summary is found', async () => {
     const caseDetail = MockData.getCaseDetail();
 
     jest.spyOn(CaseManagement.prototype, 'getCaseSummary').mockResolvedValue(caseDetail);
-    const response = await controller.getCaseSummary(applicationContext, caseDetail.caseId);
-    expect(response).toEqual({ success: true, body: caseDetail });
+    const request = mockCamsHttpRequest({ params: { caseId: caseDetail.caseId } });
+    const response = await controller.getCaseSummary(applicationContext, request);
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts:22]: again here.  The request should be in the context.

File: backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts [hunk 37]
```diff
@@ -1,40 +1,46 @@
...
   test('should throw NotFound error if case summary is not found', async () => {
     const error = new NotFoundError('CASE-MANAGEMENT-USE-CASE', {
       message: 'Case summary not found for case ID.',
     });
     jest.spyOn(CaseManagement.prototype, 'getCaseSummary').mockRejectedValue(error);
-    await expect(controller.getCaseSummary(applicationContext, '000-00-00000')).rejects.toThrow(
-      error,
-    );
+    const request = mockCamsHttpRequest({ params: { caseId: '000-00-00000' } });
+    await expect(controller.getCaseSummary(applicationContext, request)).rejects.toThrow(error);
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts:36]: again here. The request should be in the context.

File: backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts [hunk 38]
```diff
@@ -1,40 +1,46 @@
...
   });
 
   test('should throw any other error', async () => {
     const error = new Error('TestError');
     jest.spyOn(CaseManagement.prototype, 'getCaseSummary').mockRejectedValue(error);
-    await expect(controller.getCaseSummary(applicationContext, '000-00-00000')).rejects.toThrow(
+    const request = mockCamsHttpRequest({ params: { caseId: '000-00-00000' } });
+    await expect(controller.getCaseSummary(applicationContext, request)).rejects.toThrow(
```
Comment (fmaddenflx) [inline backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts:43]: again here. The request should be in the context.

File: backend/functions/cases/cases.function.test.ts (hunk 21 above)
Comment (jamesobrooks) [reply backend/functions/cases/cases.function.test.ts:112]: Screenshot 2024-09-04 at 3 29 16 PM

File: backend/functions/case-assignments/case.assignment.function.test.ts (hunk 15 above)
Comment (jamesobrooks) [reply backend/functions/case-assignments/case.assignment.function.test.ts:84]: Yes, it is what is being printed so that the test name makes it clear which case is failing if it ever fails.

File: backend/functions/orders/orders.function.ts (hunk 31 above)
Comment (fmaddenflx) [reply backend/functions/orders/orders.function.ts:35]: um... is there any difference between context.request.params['id'] and context.request.params.id?  Isn't that literally the same thing?  Or is params an actual array?

File: common/src/api/http-status-codes.ts [hunk 39]
```diff
@@ -0,0 +1,8 @@
+export const HttpStatusCodes = {
+  OK: 200,
+  CREATED: 201,
+  ACCEPTED: 202,
+  NO_CONTENT: 204,
+} as const;
```
Comment (fmaddenflx) [inline common/src/api/http-status-codes.ts:6]: Are we intending to use this only for 20X codes or do we plan to use this for 30x, 40x, and 50x?

File: backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts (hunk 36 above)
Comment (btposey) [reply backend/functions/lib/controllers/case-summary/case-summary.controller.test.ts:22]: Yes, but in this case the requests are overiden from the default request we would get back from the application context. Not changing.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx (hunk 33 above)
Comment (fmaddenflx) [inline user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:233]: We're skipping a lot of tests in this file.  We should determine if we can throw them away or if they need to be fixed.

File: user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx
Comment (jamesobrooks) [reply user-interface/src/data-verification/consolidation/ConsolidationOrderAccordion.test.tsx:1]: Keeping for now.

File: common/src/api/http-status-codes.ts (hunk 39 above)
Comment (jamesobrooks) [reply common/src/api/http-status-codes.ts:6]: We will do so, soon.

File: backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts (hunk 1 above)
Comment (btposey) [reply backend/functions/lib/controllers/attorneys/attorneys.controller.test.ts:2]: Punting for now.

File: backend/functions/orders-sync/orders-sync.function.ts (hunk 18 above)
Comment (btposey) [reply backend/functions/orders-sync/orders-sync.function.ts:10]: Punting for now.

**Review by btposey — APPROVED** [review]

**Review by fmaddenflx — APPROVED** [review]
